tracee-rules/output.go

```diff
@@ -44,24 +47,53 @@ func setupOutput(resultWriter io.Writer, clock Clock, webhook string) (chan type
 			}
 
 			if webhook != "" {
-				payload, err := prepareJSONPayload(res)
-				if err != nil {
-					log.Printf("error preparing json payload for %v: %v", res, err)
-					continue
+				if err := sendToWebhook(res, webhook, webhookTemplate, realClock{}); err != nil {
+					log.Println(err)
 				}
-				resp, err := http.Post(webhook, "application/json", strings.NewReader(payload))
-				if err != nil {
-					log.Printf("error calling webhook for %v: %v", res, err)
-					continue
-				}
-				resp.Body.Close()
 			}
 		}
 	}()
 	return out, nil
 }
 
-func prepareJSONPayload(res types.Finding) (string, error) {
+func sendToWebhook(res types.Finding, webhook string, webhookTemplate string, clock Clock) error {
+	var payload string
+
+	switch {
+	case webhookTemplate != "":
+		t, err := template.New(filepath.Base(webhookTemplate)).
+			Funcs(map[string]interface{}{
+				"unixToRFC3339": func(unixTs float64) string {
+					return time.Unix(int64(unixTs), 0).UTC().Format("2006-01-02T15:04:05Z")
+				},
+			}).ParseFiles(webhookTemplate)
+		if err != nil {
+			return fmt.Errorf("error preparing webhook template: %v", err)
+		}
+
+		buf := bytes.Buffer{}
+		if err := t.Execute(&buf, res); err != nil {
+			return fmt.Errorf("error writing to the template: %v", err)
+		}
+		payload = buf.String()
+
+	default:
+		var err error
+		payload, err = prepareJSONPayload(res, clock)
```

Comment:
I left the default behaviour as is to keep it backwards compatible. Are you suggesting we make a new template type for falcosidekick? If so what would be the default behaviour be in that case? A "simpler" JSON payload like the following?
```
{
	"Timestamp": "2021-02-27T07:19:07Z",
	"ProcessName": "foobar.exe",
	"HostName": "foobar.local"
}
```

Reply:
the default behavior will remain the same (send a json payload compatible with falco), but the implementation will be different as to generate the json using a default template instead of the falcosidekick's Payload struct

Reply:
Let's do it as a follow up PR to track this https://github.com/aquasecurity/tracee/issues/592